### PR TITLE
delete timeseries when path is not found

### PR DESCRIPTION
**Pull request by Ring-k**
Problem >>> If the DeleteTimeseriesPlan is routed globally and the metadata is removed, and the other node needs to retrieve the full paths, then the metadata is nowhere to be found.

Solution >>> In this implementation, the full paths will be retrieved before plan routing.

### CODE REVIEW

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java [hunk 1]
```diff
@@ -2614,6 +2641,43 @@ private AsyncClientPool getDataAsyncClientPool() {
...
+        .newFixedThreadPool(partitionTable.getGlobalGroups().size());
+    for (String pathStr : originalPaths) {
+      getAllPathsService.submit(() -> {
+        try {
+          List<String> fullPathStrs = getMatchedPaths(pathStr);
+          if (fullPathStrs.isEmpty()) {
+            nonExistPaths.add(pathStr);
+            logger.error("Path {} is not found.", pathStr);
+          }
+          fullPaths.addAll(fullPathStrs);
```
Comment (jt2594838) [inline cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java:2664]: `fullPaths.addAll(fullPathStrs);` may be put into the else branch to reduce unnecessary method calls.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ClearCachePlan.java [hunk 2]
```diff
@@ -34,4 +34,9 @@ public ClearCachePlan() {
   public List<Path> getPaths() {
     return new ArrayList<>();
   }
+
+  @Override
+  public List<String> getPathsStrings() {
+    return new ArrayList<>();
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ClearCachePlan.java:40]: Use Collections.emptyList().

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AlterTimeSeriesPlan.java [hunk 3]
```diff
@@ -102,6 +103,15 @@ public String getAlias() {
     return Collections.singletonList(path);
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    List<String> ret = new ArrayList<>();
+    for (Path curPath : Collections.singletonList(path)) {
+      ret.add(curPath.getFullPath());
+    }
+    return ret;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AlterTimeSeriesPlan.java:112]: Just `Return Collections.singletonList(path.getFullPath());` and check other places, please.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/DeletePartitionPlan.java [hunk 4]
```diff
@@ -41,6 +41,11 @@ public DeletePartitionPlan(String storageGroupName, Set<Long> partitionId) {
     return null;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    return null;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/crud/DeletePartitionPlan.java:46]: It is not a good idea to return a null collection, use `Collections.emptyList()` instead.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertRowPlan.java [hunk 5]
```diff
@@ -196,13 +196,21 @@ public void markFailedMeasurementInsertion(int index, Exception e) {
...
       ret.add(new Path(deviceId, m));
     }
     return ret;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    List<String> ret = new ArrayList<>();
+    for (String m : measurements) {
+      ret.add(new Path(deviceId, m).getFullPath());
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertRowPlan.java:209]: Just use `ret.add(deviceId + IoTDBConstant.PATH_SEPARATOR + m)`.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertTabletPlan.java [hunk 6]
```diff
@@ -125,6 +125,23 @@ public void setRange(List<Integer> range) {
...
+    if (paths != null) {
+      List<String> ret = new ArrayList<>();
+      for (Path path : paths) {
+        ret.add(path.getFullPath());
+      }
+      return ret;
+    }
+    List<Path> ret = new ArrayList<>();
+    for (String m : measurements) {
+      ret.add(new Path(deviceId, m));
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertTabletPlan.java:139]: Just use `ret.add(deviceId + IoTDBConstant.PATH_SEPARATOR + m)` and return ret.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AuthorPlan.java [hunk 7]
```diff
@@ -254,6 +254,15 @@ public String toString() {
     return ret;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    List<String> ret = new ArrayList<>();
+    if (nodeName != null) {
+      ret.add(nodeName.getFullPath());
+    }
+    return ret;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AuthorPlan.java:263]: `return nodeName != null ? Collections.singletonList(nodeName.getFullPath()) : Collections.emptyList()`.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateSnapshotPlan.java [hunk 8]
```diff
@@ -35,4 +35,9 @@ public CreateSnapshotPlan() {
   public List<Path> getPaths() {
     return new ArrayList<>();
   }
+
+  @Override
+  public List<String> getPathsStrings() {
+    return new ArrayList<>();
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateSnapshotPlan.java:41]: Use Collections.emptyList().

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateTimeSeriesPlan.java [hunk 9]
```diff
@@ -133,12 +134,21 @@ public String toString() {
...
     return Collections.singletonList(path);
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    List<String> ret = new ArrayList<>();
+    for (Path curPath : Collections.singletonList(path)) {
+      ret.add(curPath.getFullPath());
+    }
+    return ret;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateTimeSeriesPlan.java:149]: `Collections.singletonList(path.getFullPath())`

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/DataAuthPlan.java [hunk 10]
```diff
@@ -49,6 +49,11 @@ public DataAuthPlan(OperatorType operatorType, List<String> users) {
     return null;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    return null;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/DataAuthPlan.java:54]: `Collections.emptyList()`

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadConfigurationPlan.java [hunk 11]
```diff
@@ -115,6 +115,11 @@ public void deserialize(ByteBuffer buffer) {
     return null;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    return null;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadConfigurationPlan.java:120]: `Collections.emptyList()`

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadDataPlan.java [hunk 12]
```diff
@@ -48,6 +48,15 @@ public LoadDataPlan(String inputFilePath, String measureType) {
     return ret;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    List<String> ret = new ArrayList<>();
+    if (measureType != null) {
+      ret.add(measureType);
+    }
+    return ret;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadDataPlan.java:57]: Here too.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/MergePlan.java [hunk 13]
```diff
@@ -38,4 +38,9 @@ public MergePlan() {
   public List<Path> getPaths() {
     return new ArrayList<>();
   }
+
+  @Override
+  public List<String> getPathsStrings() {
+    return new ArrayList<>();
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/MergePlan.java:44]: And here.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/OperateFilePlan.java [hunk 14]
```diff
@@ -54,6 +54,11 @@ public OperateFilePlan(File file, File targetDir, OperatorType operatorType) {
     return null;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    return null;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/OperateFilePlan.java:59]: And there.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetStorageGroupPlan.java [hunk 15]
```diff
@@ -57,6 +58,15 @@ public void setPath(Path path) {
     return ret;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    List<String> ret = new ArrayList<>();
+    if (path != null) {
+      ret.add(path.getFullPath());
+    }
+    return ret;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetStorageGroupPlan.java:67]: And here.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetTTLPlan.java [hunk 16]
```diff
@@ -54,6 +54,11 @@ public SetTTLPlan(String storageGroup) {
     return null;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    return null;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetTTLPlan.java:59]: And here.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ShowPlan.java [hunk 17]
```diff
@@ -39,6 +39,11 @@ public ShowPlan(ShowContentType showContentType){
     return null;
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    return null;
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ShowPlan.java:44]: And here.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/TracingPlan.java [hunk 18]
```diff
@@ -38,6 +38,11 @@ public TracingPlan(boolean isTracingOn) {
     return new ArrayList<>();
   }
 
+  @Override
+  public List<String> getPathsStrings() {
+    return new ArrayList<>();
```
Comment (jt2594838) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/sys/TracingPlan.java:43]: And here.

File: cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java (hunk 1 above)
Comment (Ring-k) [reply cluster/src/main/java/org/apache/iotdb/cluster/server/member/MetaGroupMember.java]: You're right. Thanks for your suggestion :)

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ClearCachePlan.java (hunk 2 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ClearCachePlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AlterTimeSeriesPlan.java (hunk 3 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AlterTimeSeriesPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/DeletePartitionPlan.java (hunk 4 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/crud/DeletePartitionPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertRowPlan.java (hunk 5 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertRowPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertTabletPlan.java (hunk 6 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/crud/InsertTabletPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AuthorPlan.java (hunk 7 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/AuthorPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateSnapshotPlan.java (hunk 8 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateSnapshotPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateTimeSeriesPlan.java (hunk 9 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/CreateTimeSeriesPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/DataAuthPlan.java (hunk 10 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/DataAuthPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadConfigurationPlan.java (hunk 11 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadConfigurationPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadDataPlan.java (hunk 12 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadDataPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/MergePlan.java (hunk 13 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/MergePlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/OperateFilePlan.java (hunk 14 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/OperateFilePlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetStorageGroupPlan.java (hunk 15 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetStorageGroupPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetTTLPlan.java (hunk 16 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/SetTTLPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ShowPlan.java (hunk 17 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/ShowPlan.java]: I'll change it. Thanks for your suggestion.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/sys/TracingPlan.java (hunk 18 above)
Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/TracingPlan.java]: I'll change it. Thanks for your suggestion.

Comment (Ring-k) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/sys/TracingPlan.java]: I'll change it. Thanks for your suggestion.

**Pull request merged**